dbapi/src/main/java/com/griddynamics/jagger/dbapi/DatabaseServiceImpl.java

```diff
@@ -1260,6 +1265,65 @@ public SessionInfoProvider getSessionInfoService() {
     }
 
     @Override
+    public DecisionPerSessionDto getDecisionPerSession(String sessionId) {
+        DecisionPerSessionEntity sessionDecision = (DecisionPerSessionEntity)
+                entityManager.createQuery("select dps from DecisionPerSessionEntity as dps where dps.sessionId = :sessionId")
+                        .setParameter("sessionId", sessionId).getSingleResult();
+
+        List<DecisionPerTaskEntity> taskDecisions = (List<DecisionPerTaskEntity>)
+                entityManager.createQuery("select dpt from DecisionPerTaskEntity as dpt where dpt.taskData.sessionId = :sessionId")
+                        .setParameter("sessionId", sessionId).getResultList();
+
+        Set<Long> taskIds = taskDecisions.stream().map(decision -> decision.getTaskData().getId()).collect(Collectors.toSet());
+        Map<Long, Set<Long>> testIdsByTestGroupIds = getTestGroupIdsByTestIds(taskIds);
+
+        List<DecisionPerTestGroupDto> decisionPerTestGroupDtos = new ArrayList<>();
```

Comment:
What for? emptyList() returns immutable list and here I need simple empty ArrayList to fill it.